runtime/src/main/cpp/Porting.h

```diff
@@ -94,6 +99,23 @@ do {                          \
 } while(0)
 #endif
 
+template <typename F>
+// TODO: Use invoke_result without && when compiled with C++17
+ALWAYS_INLINE typename std::result_of<F&&()>::type tryOrTerminate(F&& tryClause) {
+#if KONAN_NO_EXCEPTIONS
+  // This will terminate the program if exception is thrown.
+  return tryClause();
+#else
+  try {
+    return tryClause();
+  } catch (...) {
```

Comment:
Does this additional `catch` impose performance penalty?

Reply:
No measurable difference according to `objcinterop` benchmarks. However, I do worry about having an extra `try-catch` on a happy path.

Reply:
> having an extra `try-catch` on a happy path.

Exactly! That's what I meant.

Reply:
> Clang and gcc follow zero-cost Itanium ABI

What about ABI on ARM? Does it do a similar zero-cost handling?